app/preprints/-components/preprint-provider-selection/component.ts

```diff
@@ -0,0 +1,44 @@
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { serviceLinks } from 'ember-osf-web/const/service-links';
+import { tracked } from '@glimmer/tracking';
+import Component from '@glimmer/component';
+import PreprintProviderModel from 'ember-osf-web/models/preprint-provider';
+import RouterService from '@ember/routing/router-service';
+
+
+interface InputArgs {
+    allProviders: PreprintProviderModel[];
+}
+
+
+export default class PreprintProviderSelection extends Component<InputArgs> {
+    @service router!: RouterService;
+
+    allProviders: PreprintProviderModel[] = this.args.allProviders;
+    learnMoreUrl: string = serviceLinks.preprintsSupport;
+    @tracked selectedProvider: PreprintProviderModel | undefined;
+
+    public get isDisabled(): boolean {
+        return this.selectedProvider === undefined;
+    }
```

Comment:
Nothing wrong with what is written here, but just wanted to point out some Javascript/Typescript shorthand here you could use.

```
    @tracked selectedProvider?: PreprintProviderModel;

    public get isDisabled(): boolean {
        return Boolean(this.selectedProvider);
    }
```
where the [optional property](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean#boolean_coercion) symbol `?` indicates that this tracked variable may be undefined (and by declaring it without setting it to anything, it will be `undefined`), and the `Boolean` object here converts any value to either truthy or falsey (some weird Javascript boolean pitfalls to be weary of can be found here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean#boolean_coercion)